### ISSUE TITLE
GradCAM on freesolv dataset

### ISSUE THREAD

**Issue by boma-dirgantara** [divelab/DIG#72]
Hi DIG maintainers,

I have tried to train GIN_3l on freesolv dataset with regression task. But it seems the model will not learn. So my question is, can I use freesolv dataset for GradCAM? if yes could you share the checkpoint for GIN_3l model?

Thanks in advance,
Boma

**Comment by Oceanusity** [comment]
I haven't try the `freesolv` dataset yet, and I think I won't provide such checkpoint recently. 

As an advance, how about trying [gradient clipping](https://pytorch.org/docs/stable/generated/torch.nn.utils.clip_grad_norm_.html) during the training?

**Comment by boma-dirgantara** [comment]
Okay thanks @Oceanusity the model is now learned.

If I may post the following question. As I train using regression task, I assume `num_classes=1` and for `collect_data` become  `x_collector.collect_data(masks, related_preds)` in [cell 7](https://github.com/divelab/DIG/blob/dig/examples/xgraph/gradcam.ipynb) so that it not trigger `key, value in related_preds[label].items():` in [line 158](https://github.com/divelab/DIG/blob/dig/dig/xgraph/evaluation/metrics.py#L157-L164). But all result of related_prediction give the same number for `'zero': 1.0, 'masked': 1.0, 'maskout': 1.0, 'origin': 1.0`. Is it true that the graph not being explained? because as I run further the example code the `Fidelity and Fidelity_inv give both the result 0.0000`

**Comment by Oceanusity** [comment]
I don't get the idea. In fact, all the methods are applied on classification problem now.

**Comment by boma-dirgantara** [comment]
I see. Thanks for the Info @Oceanusity . I thought because the target value from `freesolv` dataset is floating number it should use regression task
